src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -358,23 +370,124 @@ def setClean(self, wflow):
         wflow['IsClean'] = self._checkClean(wflow)
         return wflow
 
+    def _checkLogDBClean(self, wflow):
+        """
+        An auxiliary function used to only check the LogDB cleanup status.
+        It makes a query to LogDB in order to verify there are no any records for
+        the current workflow
+        :param wflow:       A MSRuleCleaner workflow representation
+        :return:            Bool
+        """
+        cleanStatus = False
+        logDBRecords = self.logDB.get(wflow['RequestName'])
+        self.logger.debug("logDBRecords: %s", pformat(logDBRecords))
+        if logDBRecords:
+            cleanStatus = True
+        return cleanStatus
+
+    def setLogDBClean(self, wflow):
+        """
+        A function to set the 'IsLogDBClean' flag based on the presence of any
+        records in LogDB for the current workflow.
+        :param  wflow:      A MSRuleCleaner workflow representation
+        :return wflow:
+        """
+        wflow['IsLogDBClean'] = self._checkLogDBClean(wflow)
+        return wflow
+
+    def findTargStatus(self, wflow):
+        """
+        Find the proper targeted archival status
+        :param  wflow:      A MSRuleCleaner workflow representation
+        :return wflow:
+        """
+        # Check the available status transitions before we decide the final status
+        targetStatusList = RequestStatus.REQUEST_STATE_TRANSITION.get(wflow['RequestStatus'], [])
+        for status in targetStatusList:
+            if self.targetStatusReg.match(status):
+                wflow['TargetStatus'] = status
+        self.logger.debug("TargetStatus: %s", wflow['TargetStatus'])
+        return wflow
+
+    def _cleanLogDB(self, wflow):
+        """
+        A function to be used for cleaning all the records related to a workflow in logDB.
+        :param wflow:       A MSRuleCleaner workflow representation
+        :return:            Bool
+        """
+        cleanStatus = False
+        try:
+            self.logger.info("Deleting %s from LogDB WMStats...", wflow['RequestName'])
+            res = self.logDB.delete(wflow['RequestName'], agent=False)
+            if res == 'delete-error':
+                self.logger.error("Failed to delete logDB docs for wflow: %s", wflow['RequestName'])
+                return cleanStatus
+            self.setLogDBClean(wflow)
```

Comment:
So in this case we do not fetch the document from a couchdb view, but from the logDb instance itself, and that's  why I was expecting the change to be kind of instant. The additional check was done only because I did not want to have some leftovers behind, but I agree - In case we are  sure that once the deletion call is properly sent then the documents are 100% about to be deleted, then yes this additional call to the database would be redundant.

Reply:
BTW I am about to make the change you requested here. Lets see how it goes.

Reply:
Done